STL_Extension/include/CGAL/Compact_container.h

```diff
@@ -1109,9 +1110,21 @@ namespace internal {
   std::size_t hash_value(const CC_iterator<DSC, Const>&  i)
   {
     typedef Time_stamper_impl<typename DSC::value_type> Stamper;
-    return Stamper::hash_value(&*i);
+    return Stamper::hash_value(i.operator->());
```

Comment:
This seems a valid fix to me, as [`Stamper::hash_value()`](https://github.com/CGAL/cgal/blob/master/STL_Extension/include/CGAL/Time_stamper.h#L84) checks if the pointer is a null pointer.